pkg/server/nodemanager/node_scaler.go

```diff
@@ -84,10 +85,17 @@ func (s *BindingNodeScaler) createNodeForPod(pod *api.Pod) *api.Node {
 	if s.bootLimiter.IsUnavailableInstance(pod.Spec.InstanceType, isSpotPod) {
 		return nil
 	}
-
+	// XXX henry: we need the instance’s architecture to be able to pick the right image.
+	// This is currently handled by CloudClient, which isn’t accessible from this object.
+	// We assume x84_64 because everybody still uses it.
+	image, found := BootImages[cloud.Arch_x86_64]
```

Comment:
How is this going to work for mac1 instances?

Reply:
See my comment on using the instanceselector package. We can do the instance family -> architecture mapping there.